subsys/bluetooth/host/l2cap.c

```diff
@@ -3000,15 +3012,21 @@ void l2cap_tx_destroy(void *meta_data)
 		struct l2cap_tx_meta_data *data = meta_data;
 
 		if (data->user_data) {
-			att_tx_destroy(data->user_data);
+			do_destroy(data->user_data);
 		}
 
 		free_tx_meta_data(data);
 
-		return;
+		return true;
 	}
 #endif /* CONFIG_BT_L2CAP_DYNAMIC_CHANNEL */
 
 	/* Not L2CAP metadata, let ATT handle it */
-	att_tx_destroy(meta_data);
+	do_destroy(meta_data);
```

Comment:
Wouldn't it be better to remove this line? Any destructor that would be called here could instead be registered with `conn_userdata_destroy`.